src/main/cpp/Logging.cpp

```diff
@@ -0,0 +1,157 @@
+#include "Logging.hpp"
+
+namespace wcr::logging
+{
+    bool Logger::checkLevel()
+    {
+        const auto log = (m_global_level & m_level) != 0;
+
+        //wpi::outs() << "Is Valid Level: " << std::to_string(log) << "\n";
+        //wpi::outs() << "Global Level: " << std::to_string(m_global_level) << ", Listening Level: " << std::to_string(m_level) << "\n";
+        //wpi::outs().flush(); 
+
+        if (log && m_new_line && m_construct_name) {
+            wpi::outs() << "[" << constructName() << "]: ";
```

Comment:
Aside from adding the log levels, it could be cool to assign different text colors based on log level. I'm not sure if that can be done, but it could be cool. We may need to research the [API](https://github.wpilib.org/allwpilib/docs/release/cpp/classwpi_1_1raw__fd__ostream.html) to see if that's possible.

Reply:
Might be possible, but something for the future